### PR TITLE
Automate release artifacts build: usage pytorch cxx11 builder base image

### PR DESCRIPTION
**Pull request by lanluo-nvidia**
# Description

Added two github workflows for the automate release artifacts:

release-linux:
4 auditwheel repaired wheel (python version: 3.8/3.9/3.10/3.11, cuda version: cu124)
1 pre-cxx11-abi tarball (python version: 3.10, cuda version: cu124)
1 cxx11-abi tarball (python version: 3.10, cuda version: cu124)
Here is the 2.4.0 linux release artifacts: https://github.com/pytorch/TensorRT/actions/runs/9816517467#artifacts
 https://github.com/pytorch/TensorRT/actions/runs/9860448640
 
release-windows:
4 wheel file(python version: 3.8/3.9/3.10/3.11, cuda version: cu124)
Here is the 2.4.0 windows release artifacts:
https://github.com/pytorch/TensorRT/actions/runs/9850845389#artifacts

## Type of change

Please delete options that are not relevant and/or add your own.

- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality to not work as expected)
- This change requires a documentation update

# Checklist:

- [ ] My code follows the style guidelines of this project (You can use the linters)
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas and hacks
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests to verify my fix or my feature
- [ ] New and existing unit tests pass locally with my changes
- [ ] I have added the relevant labels to my PR in so that relevant reviewers are notified

### CODE REVIEW

File: .github/scripts/generate-release-matrix.py [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+import argparse
+import json
+import sys
+
+RELEASE_CUDA_VERSION = {
+    "wheel": ["cu124"],
+    "tarball": ["cu124"],
+}
+RELEASE_PYTHON_VERSION = {
+    "wheel": ["3.8", "3.9", "3.10", "3.11"],
```
Comment (narendasan) [inline .github/scripts/generate-release-matrix.py:12]: we need 3.12 for 2.4 as well

File: .github/workflows/release-wheel-linux.yml [hunk 2]
```diff
@@ -0,0 +1,259 @@
...
+      - name: Build the wheel (bdist_wheel)
+        working-directory: ${{ inputs.repository }}
+        shell: bash -l {0}
+        run: |
+          set -x
+          source "${BUILD_ENV_FILE}"
+
+          # BUILD_VERSION example: 2.4.0+cu121, we don't want the +cu121 part, so remove +cu121
+          BUILD_VERSION=${BUILD_VERSION%+*}
+          TRT_VERSION=10.0.1
```
Comment (narendasan) [inline .github/workflows/release-wheel-linux.yml:197]: Is there a way to have this pulled in from either dev_dep_versions or another source in the codebase?

File: .github/workflows/release-wheel-linux.yml [hunk 3]
```diff
@@ -0,0 +1,259 @@
...
+      # NB: Only upload to GitHub after passing smoke tests
+      - name: Upload wheel to GitHub
+        if: ${{ inputs.cxx11-tarball-release != 'true' }}
+        continue-on-error: true
+        uses: actions/upload-artifact@v3
+        with:
+          name: ${{ env.ARTIFACT_NAME }}
+          path: ${{ inputs.repository }}/release/wheel/
+      - name: Upload pre-cxx11 tarball to GitHub
+        if: ${{ inputs.cxx11-tarball-release != 'true' && env.PYTHON_VERSION == '3.10' && env.CU_VERSION == 'cu124' }}
```
Comment (narendasan) [inline .github/workflows/release-wheel-linux.yml:243]: What is this check for?

Comment (lanluo-nvidia) [reply .github/workflows/release-wheel-linux.yml:243]: release-wheel-linux.yml is shared by release wheel and tarball:
1. release wheel uses python version: 3.8---3.12
2. release tarball is only for python 3.10
this check is for release pre-cxx11 tarball:
env.CU_VERSION == 'cu124' is not actually required, let me remove it.

**Review by narendasan — APPROVED** [review]
Lgtm!

**Pull request merged**